docs/source/apps/hpc.md

```diff
@@ -0,0 +1,33 @@
+# HPC integration
+
+## Interact with HPC through MarketPlace proxy using HPC gateway SDK
+
+The HPC gateway SDK is provided for the app developers or the MarketPlace users to run the time consuming tasks over the computational clusters.
+
+For details of how to use SDK to interact with HPC cluster deployed, read the section [here](../jupyter/hpc-sdk.ipynb)
+
+The following methods are supported by the SDK to interact with cluster over MarketPlace proxy.
+
+- Check the availability of system: `app.heartbeat()`
+- Create a new user: `app.create_user()`. Only when user first time use the functionality.
+- Create a new calculation: `app.create_job()`, the jobid will returned for further job preparation.
+- Show job state (List files in job folder): `app.check_job_state(jobid=<jobid>)`, list files in remote job folder.
+- Upload file: `app.upload_file(jobid=<jobid>, filename=<filename>, source_path=<local_file_path>`, upload file to remote folder from local path.
+- Download file: `app.download_file(jobid=<jobid>, filename=<filename>`, download file from remote folder.
+- Delete file: `app.delete_file(jobid=jobid, filename=<filename>)`, delete the file from remote folder.
+- Launch job: `app.launch_job(jobid=<jobid>)`, launch/submit the job to cluster queue managed by slurm.
+- Cancel the job: `app.cancel_job(jobid=jobid)`, cancel the job submitted.
+
+## The architecture of user folder
+
+The first time a user needs to call the `app.create_user()` function to create a user in the HPC gateway app's database and create a folder on the remote cluster to store calculations.
+The remote user folder is named based on the string of the user's Marketplace Email before the `@` symbol, with all non-letter characters converted to underscores.
```

Comment:
so `pablo@fraunhofer.de` and `pablo@epfl.at` would have the same folder?

Reply:
Yes, that is the case, probably better to convert the whole email to the folder name, including `@` and `.` converted to `_`?

Reply:
What information do you have from the user? (i.e., where do you get the e-mail from?) A unique id might be more elegant if available.

Reply:
The information is read from user_info, there is also sub_id that can use, but not very meaningful.